torch/distributed/_tensor/ops/math_ops.py

```diff
@@ -280,6 +282,111 @@ def softmax_bwd_rule(op_schema: OpSchema) -> OutputSharding:
     return pointwise_rule(op_schema)
 
 
+@register_op_strategy(
+    [aten.nll_loss_forward.default, aten.nll_loss2d_forward.default],
+    schema_info=RuntimeSchemaInfo(3),
+)
+def nll_loss_forward_strategy(mesh: DeviceMesh, op_schema: OpSchema) -> OpStrategy:
+    assert len(op_schema.args_schema) == 5
+    (
+        input_strategy,
+        target_strategy,
+        weight_strategy,
+        reduction,
+        _,
+    ) = op_schema.args_schema
```

Comment:
question: what is the argument list of this operator?
Is this the associated operator? https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/native_functions.yaml#L11374-L11377

If so, why is the code (https://github.com/pytorch/pytorch/blob/main/torch/_refs/nn/functional/__init__.py#L652-L658) has argument `reduction` of type `str`?
cc @wanchaol

Reply:
iiuc the computation happens in C++ at
https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/LossNLL.cpp#L678

and the code pointers should be 
https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/native_functions.yaml#L11392
https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/native_functions.yaml#L11423

not sure when the PyTorch decompositions are called...

Reply:
nn.functional have potentially different signatures with the aten op, what we should focus on is the aten op instead as that's where we overwrite the ops, the nn.functional would under the hood decompose to a set of aten ops if necessary

However we should test against nn.functional as that's the most user is using